### ISSUE TITLE
Null flags datapoints are not ignored

### ISSUE THREAD

**Issue by mitar** [highslide-software/highcharts.com#4487]
See [reproduction](http://jsfiddle.net/nao92j9w/).

You get `Highcharts error #15: www.highcharts.com/errors/15` error.

This is strange, because for other chart types null datapoints are simply ignored.

**Comment by pawelfus** [comment]
The problem is that 'null' is at the last index in data array, so Highcharts initialize that point with starting configuration: `x: 0`. Adding more null-points at the end, will increment `x+=1`. That comes from default options: `pointStart` and `pointInterval`. 

Demo: http://jsfiddle.net/b2wg6uq5/ (the same error, different series type).

**Comment by mitar** [comment]
This is definitely surprising. Because if data is coming from the server, it is tricky do detect that errors are just when there are nulls at the end.

Is it possible to disable this behavior?

**Comment by pawelfus** [comment]
First of all, I would ask why you need null-point flag? Wouldn't be better to filter out unnecessary points?

**Comment by mitar** [comment]
Oh, just so that it is consistent with the rest. So I have a wrapper around highcharts to render various time-series data. Like a dashboard. You select which series you want, which timespan. And then it renders. And data can contain null values normally (to force breaking the line when drawing). So data has null values. And which type of visualization is used to render a particular data is orthogonal to the data representation. At least I would like to keep it that way. Additionally, you are saying that this issue happens even for non-flag series types. So this is the issue. If there null points somewhere in the middle of the data and user selects a timespan which ends in the middle of that region with nulls, there will be issues?

I could probably check this special case every time I read the data from the server, but it does not feel a correct way to do things. Like if there are null values at the end, remove them. But otherwise keep them.

**Comment by pawelfus** [comment]
Sorry for not being unclear and misleading. It doesn't matter where null is placed in fact (example: http://jsfiddle.net/b2wg6uq5/2/ )  - just when you set points with x-y pairs, then such null point should be set as x-null, not null object. 

Consider these two examples:

 1. Data is set as: `[10, 20, 50, null, 10, 20, null, 10]` - it will place null's in a propel place because **all** of the points are using `pointStart` and `pointInterval` options.

 2. Data is set as: `[ [1, 10], [10, 20], [10000, 50], null, [10001, 10], [20000, 20], null, [50000, 10] ]` (or the same but in object notation). Now null points will **still** use `pointInterval` and `pointStart`, while other points will come with `x` value already. It's really hard to determine if first null should be placed at x=10000.5 or maybe x=10000.000001 ? Not even talking about next null. For some users placing in the middle is perfectly fine, but for other it should be next number, like 20001. In such case those nulls should be set this way: `[ [1, 10], [10, 20], [10000, 50], [10000.01, null], [10001, 10], [20000, 20], [20000.01, null], [50000, 10] ]. 

So in other words, solution for your case is to provide nut empty/null points, but provide at least x-position for those null-point flags.

Let me know if something is still unclear for you.

**Comment by pawelfus** [comment]
Though thing, to consider: flags require only `x` value to be displayed on the chart, so there is no real way to 'nullify' flag points. I'm still curious why in the first place null-flag should exist..

**Comment by mitar** [comment]
Ah, I see. In fact I was using `[x, null]` for non-flag datapoints. So this is why it worked. But for flags I was passing `null` directly.

OK, then I will keep my current approach where I filter out `null`s myself. But yea, I do have data with X coordinate and `null` value for flags. It is simply how data comes in. The reason is that we are downsampling the data on the server. So if you zoom out, you get precomputed aggregated data for each timespan. And when aggregating if there were no events (flags), value is `null`, but there is still a time value, because we know in which interval there were no events.

Now I need a special case for flags, but I need that anyway because for other types I use `[X, value]` format for data, and for events I use `{x: X, title, text}` format.
